enterprise_access/apps/api/v1/views.py

```diff
@@ -746,7 +746,128 @@ def partial_update(self, request, *args, **kwargs):
         return response
 
 
-class SubsidyAccessPolicyViewset(PermissionRequiredMixin, viewsets.GenericViewSet):
+class SubsidyAccessPolicyCRUDViewset(PermissionRequiredMixin, viewsets.ModelViewSet):
+    """
+     Viewset for Subsidy Acess Policy CRUD operations.
+     """
+
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = serializers.SubsidyAccessPolicyCRUDSerializer
+    authentication_classes = (JwtAuthentication,)
+    filter_backends = (filters.OrderingFilter, DjangoFilterBackend,)
+    filterset_fields = ('group_uuid', 'policy_type',)
+    pagination_class = PaginationWithPageCount
+    http_method_names = ['get', 'head', 'options', 'post', 'patch', 'delete']
+    lookup_field = 'uuid'
+    lookup_value_regex = '[0-9a-f-]+'
+    permission_required = 'requests.has_admin_access'
+
+    @property
+    def requested_group_uuid(self):
+        """
+        The group uuid from request params or post body
+        """
+        group_uuid = None
+        if self.action in ('retrieve', 'partial_update', 'destroy'):
+            policy_uuid = self.kwargs.get('uuid')
+            if policy_uuid:
+                try:
+                    policy = SubsidyAccessPolicy.objects.get(uuid=policy_uuid)
+                    group_uuid = str(policy.group_uuid)
+                except ObjectDoesNotExist:
+                    group_uuid = None
+        elif self.action == 'create':
+            group_uuid = self.request.POST.get('group_uuid')
+        else:
+            group_uuid = self.request.query_params.get('group_uuid', None)
+        return group_uuid
+
+    def get_permission_object(self):
+        """
+        Returns the enterprise uuid (group uuid) to verify that requesting user possess the enterprise admin role.
+        """
+        return self.requested_group_uuid
+
+    def create(self, request, *args, **kwargs):
+        """
+        create action for SubsidyAccessPolicyCRUDViewset. Handles creation of policy after validation
+        """
+        policy_data = request.data
+        serializer = self.get_serializer(data=policy_data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
this can also be written like
```python
serializer.is_valid(raise_exception=True)
serializer.save()
return Response(serializer.data, status=status.HTTP_201_CREATED)
```